### ISSUE TITLE
Consider making the main README up-to-date with the latest release

### ISSUE THREAD

**Issue by s3rvac** [avast/retdec#860]
At present, [the main README](https://github.com/avast/retdec#retdec) reflects the state of the current `master`. This is good from the development point of view. However, when there is a breaking change since the last release, users might be confused (see e.g. #859). We should consider making the main README up-to-date with the latest release.

Alternatively, we should mention that the current README corresponds to `master` and provide links to READMEs corresponding to each release. E.g.
* [v4.0](https://github.com/avast/retdec/tree/v4.0#retdec)
* [v3.3](https://github.com/avast/retdec/tree/v3.3#retdec)
* ...

**Comment by daveime68** [comment]
"At present, the main README reflects the state of the current master."

But it doesn't reflect the state of the prebuilt binaries linked on the very same page. You don't see this is a problem?

**Comment by s3rvac** [comment]
Yes, the current state is definitely not ideal, which is the reason why I have created this ticket. Until it is addressed, in commit ca3a8e93595788b0a55eaf2c44f41360c3e41c34, I tried to rewrite instructions in README to make them a bit less confusing. I am not sure if I succeeded in that, because it is hard to write instructions for both development and stable versions in the same document.

I would leave the rest for this ticket. If you have a suggestion, feel free to write it down here or submit a pull request.
